sdk/core/src/request.rs

```diff
@@ -45,6 +45,10 @@ impl Request {
         &self.headers
     }
 
+    pub fn headers_mut(&mut self) -> &mut HeaderMap {
```

Comment:
@rylev @MindFlavor or would it make more sense to keep headers in a Cell? Reading and writing them should be fluid, as it in our other languages' SDKs. Policies may read or write to them. This works and is idiomatic, but would a Cell also be okay?

Reply:
`Cell` really wouldn't be appropriate here. What speaks for the use of `Cell`? I'm not sure I follow.

Reply:
As it is constructed now, only a single `Policy` can mutate the whole request at a given time (via `&mut Request`) so I do not see the need for `Cell`. Can you please elaborate further?

Reply:
Headers should always be mutable by any policy. In fact, across our existing languages, I would say most of the policies tend to read and/or write headers. I thought a Cell (or maybe RefCell) would've provided an always-mutable reference to the HeaderMap, but I may have misunderstood it's intent.

Still, the point is that headers need to be mutable. Is adding a method like this the idiomatic way then?